tests/test_qnode.py

```diff
@@ -177,11 +198,17 @@ def test_best_method_is_device(self):
 
         dev = CustomDeviceWithDiffMethod()
 
-        res = QNode.get_best_method(dev, "jax")
-        assert res == ("device", {}, dev)
+        with pytest.raises(
+            qml.PennyLaneDeprecationWarning, match="QNode.get_best_method is deprecated"
+        ):
```

Comment:
Same for the rest of the tests.

```suggestion
        with pytest.warns(
            qml.PennyLaneDeprecationWarning, match="QNode.get_best_method is deprecated"
        ):
```